### ISSUE TITLE
Wrong implementation AsselmblyResolve 

### ISSUE THREAD

**Issue by sidatacom** [nopSolutions/nopCommerce#5076]
We identify with microsoft together that your actual implementation is not recommended:
 
```
        private Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
        {
            //check for assembly already loaded
            var assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName == args.Name);
            if (assembly != null)
                return assembly;

            //get assembly from TypeFinder
            var tf = Resolve<ITypeFinder>();
            if (tf != null)
            {
                assembly = tf.GetAssemblies().FirstOrDefault(a => a.FullName == args.Name);
                return assembly;
            }

            return null;
        }
```
The resolver should allways return null if not found the right assembly,
its not allowed and produces errors if return the requestedassembly instead of the right on.

We opened a ticket because we have a lot of problems with our plugin.
We also push a correction for the plugin implementation which is also not perfect !!!

Redards
Alexander

**Comment by skoshelev** [comment]
Hi @sidatacom.
Could you provide a more detailed according to the problem:
1. What is meant by "right assembly", since we always return either the assembly found by its full name or null
2. Please share a link or a log of a discussion of the problem with specialists from Microsoft regarding this issue
3. What kind of problems do you get when developing your plugins?

**Comment by skoshelev** [comment]
We close the ticket as there is no additional information and objective reasons to believe that there is something wrong with this code.

Closed #5076
